tools/attach_logs_in_jira.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+import sys
+import os
+import argparse
+import logging
+import tarfile
+from pathlib import Path
+from functools import partial
+from typing import Optional
+
+import jira
+import retry
+
+from consts import JIRA_PROJECT, JIRA_SUMMARY, JIRA_SERVER
+from utils import days_ago
+
+JIRA_ATTACHMENT_MAX_SIZE = 50 * 2**20  # 50MiB
+
+log = logging.getLogger(__name__)
+stream_handler = logging.StreamHandler(sys.stderr)
+stream_handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+log.addHandler(stream_handler)
+
+
+def _get_or_create_jira_ticket(jira_client: jira.JIRA, summary: str) -> jira.Issue:
+    matching_issues = jira_client.search_issues(
+        f'project = "{JIRA_PROJECT}" AND summary ~ "{summary}"', fields=["attachment"]
+    )
+
+    if len(matching_issues) == 0:
+        log.info("Creating a placeholder issue for %s", summary)
+        return jira_client.create_issue(
+            project=JIRA_PROJECT,
+            summary=summary,
+            components=[{"name": "Cloud-Triage"}],
+            priority={"name": "Blocker"},
+            issuetype={"name": "Bug"},
+            description=(
+                "Logs are still being uploaded.\nTicket description and signatures will get processed shortly..."
+            ),
+        )
+
+    if len(matching_issues) == 1:
+        log.debug("Using pre-existing issue %s", matching_issues[0])
+        return matching_issues[0]
+
+    raise RuntimeError(
+        f"Found more than one matching issue for '{summary}': {', '.join(issue.key for issue in matching_issues)}"
+    )
+
+
+def _archive_dir(directory: Path) -> Path:
+    archived_file = directory.with_suffix(".tar.gz")
```

Comment:
FYI I just learned that `with_suffix` removes the original extension. Disgusting. so if the directory ends with something like `.d` (which some directories often do) it will be deleted

Reply:
thanks!
to be on the safe side I'll handle this case